fh_altair/__init__.py

```diff
@@ -1,22 +1,57 @@
+import json
 from uuid import uuid4
-from fasthtml.common import Div, Script
 
+from fasthtml.common import Div, Script, Style
 
 altair_headers = [
     Script(src="https://cdn.jsdelivr.net/npm/vega@5"),
     Script(src="https://cdn.jsdelivr.net/npm/vega-lite@5"),
-    Script(src="https://cdn.jsdelivr.net/npm/vega-embed@6")
+    Script(src="https://cdn.jsdelivr.net/npm/vega-embed@6"),
+    Style("""
+    .vega-embed {
+      width: 100%;
+      display: flex;
+    }
+    .vega-embed details,
+    .vega-embed details summary {
+      position: relative;
+    }
+    """),
 ]
 
 
 def altair2fasthml(chart):
     """This is the version with bad spelling"""
-    print("You have imported `altair2fasthml` which is a misspelled function. Sorry about that! It will be deprecated in favour of `altair2fasthtml` in a later release.")
+    print(
+        "You have imported `altair2fasthml` which is a misspelled function. Sorry about that! It will be deprecated in favour of `altair2fasthtml` in a later release."
+    )
     return altair2fasthtml(chart)
 
 
-def altair2fasthtml(chart):
+def altair2fasthtml(chart, vega_options={"actions": False}, full_width=False):
+    """Convert an Altair chart to a FastHTML FT component
+
+    Parameters
+    ----------
+    chart : altair.Chart
+        An Altair chart
+    vega_options : dict, optional
+        Options dictionary passed to the vegaEmbed function, by default {"actions": False}
+        which hides the Vega action menu.
+    full_width: bool, optional
+        When enabled, property width="responsive" is added to the chart allowing it to take up full width of parent Div
+
+    Returns
+    -------
+    fastcore.xml.FT
+        A FastHTML FT component containing both the target div and embedding script
+
+    """
+    if full_width:
+        chart = chart.properties(width="container")
```

Comment:
Just to check, does this override existing properties on the chart?

Reply:
As in, if the height is set beforehand, does this test remove that property?

Reply:
Good question, I just went to do a little test and confirmed that the original height is preserved.

```python
def generate_chart():
    pltr = pd.DataFrame({"y": [1, 2, 3, 2], "x": [3, 1, 2, 4]})
    chart = (
        alt.Chart(pltr)
        .mark_line()
        .encode(x="x", y="y")
        .properties(width=400, height=200)
    )
    return altair2fasthtml(chart, full_width=True)

x = generate_chart()

# Results in a chart with height = 200, width = "container"
print(to_xml(x))
```

Reply:
Ah nice, could we add that as a unit test?